plugins/uefi-capsule/fu-uefi-cod-device.c

```diff
@@ -147,6 +147,39 @@ fu_uefi_cod_device_get_results(FuDevice *device, GError **error)
 	return TRUE;
 }
 
+static gchar *
+fu_uefi_cod_device_get_filename_insyde(FuUefiDevice *self, GError **error)
+{
+	g_autofree gchar *esp_path = fu_uefi_device_get_esp_path(self);
+	for (guint i = 0; i < 0xFFFF; i++) {
+		g_autofree gchar *basename = g_strdup_printf("CapsuleUpdateFile%04X.bin", i);
```

Comment:
Before making this type of change we need to confirm if this is an Insyde behavior or a Huawei behavior.  If it's a Huawei behavior we should quirk against something different.

Reply:
Or maybe even go off a combination of Insyde and Huawei as Huawei may use another IBV in the future.

Reply:
For my future self:

    # HUAWEI MateStation S [BIOS]
    [1b7f560d-2646-4232-b526-b0b1897e6dcd]
    Flags = is-insyde

which maybe we should use instead? Also, is-insyde s a bad flag name.

Reply:
We tested this mechanism on Acer and Huawei machines. We found the mechanism present in Lenovo machines as well. This is not Huawei-specific but Insyde-specific.

Reply:
These were with the fixed Insyde stack that we quirk around requiring a specific bit?  Or is this part of that original bug?

Reply:
```
# HUAWEI MateStation S [BIOS]
[1b7f560d-2646-4232-b526-b0b1897e6dcd]
Flags = is-insyde
```

How about CoDprefix = `foo`

Reply:
I do wonder if we should add those as quirk guids to the UEFI device.

Reply:
```
<firmware gtype="FuAcpiUefi">
  <flags>has-checksum</flags>
  <id>UEFI</id>
  <size>0x236</size>
  <data size="0x236">UEFI6.</data>
  <revision>0x1</revision>
  <oem_id>HUAWEI</oem_id>
  <oem_table_id>H2O BIOS</oem_table_id>
  <oem_revision>0x1</oem_revision>
  <is_insyde>true</is_insyde>
  <guid>9d4bf935-a674-4710-ba02-bf0aa1758c7b</guid>
</firmware>
```
Where is the `insyde_cod_status` key?
IE - are we trying to work around the Insyde bug?

Reply:
> are we trying to work around the Insyde bug

Nah, what maybe I'm trying to do is say "for **any** Insyde implementation of CoD use the XXXX.bin filename" -- but do you know if Lenovo is working with the fwupd-GUID.cap format too? i..e we got that from *somewhere*.

Reply:
I've never seen a Lenovo system that uses Insyde first hand.  All the ones that i've seen use Phoenix.

Reply:
Do you know what glob Phoenix uses? e.g. I know a few of the aarch64 ones use `*.cap` for example.

Reply:
Phoenix doesn't support it on those systems that I've seen

Reply:
I think we're going to need these heuristics; probably adding them when we discover them rather than in any planned way. You don't have any insight into AMI, do you?

Reply:
I've only ever seen it on Insyde.

Reply:
@apalos anything you can add from an aarch64 point of view?